src/sentry/web/api.py

```diff
@@ -492,6 +492,7 @@ def process(self, request, project, key, auth, helper, data, **kwargs):
             helper.ensure_does_not_have_ip(data)
 
         # mutates data (strips a lot of context if not queued)
+        data['key_id'] = key.id
```

Comment:
The only thing I'd really argue is that this should belong inside of the `ClientApiHelper` where other normalizations and mutations occur.

Reply:
I guess probably the best place seems to be `LazyData._decode`. You can pass along `key` to the constructor, then apply it in `_decode` similar to how `sdk` is applied.

Reply:
Agree this should probably be added before filters etc are applied. https://github.com/getsentry/sentry/blob/master/src/sentry/coreapi.py#L418 seems to be the other place where we do something similar.

Reply:
Of course if you could make it so that `sdk`, `project` and `key` were all set in the same place, that would be amazing

Reply:
👍  moved both project and key to `LazyData._decode`